### PR TITLE
Use single map method where possible.

### PR DESCRIPTION
**Pull request by jmark99**
Replace common usage patterns of java.util.Map with Java 8 methods such as computeIfAbsent(), putIfAbsent(), etc.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/mapred/AccumuloOutputFormat.java [hunk 1]
```diff
@@ -552,13 +550,9 @@ public void close(Reporter reporter) throws IOException {
...
           HashMap<String,Set<SecurityErrorCode>> tables = new HashMap<>();
-          for (Entry<TabletId,Set<SecurityErrorCode>> ke : e.getSecurityErrorCodes().entrySet()) {
+          for (var ke : e.getSecurityErrorCodes().entrySet()) {
             String tableId = ke.getKey().getTableId().toString();
-            Set<SecurityErrorCode> secCodes = tables.get(tableId);
-            if (secCodes == null) {
-              secCodes = new HashSet<>();
-              tables.put(tableId, secCodes);
-            }
+            Set<SecurityErrorCode> secCodes = tables.computeIfAbsent(tableId, k -> new HashSet<>());
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/mapred/AccumuloOutputFormat.java]: One of the great things about `.computeIfAbsent()` is that you don't need to assign it to a variable... you could just immediately use it on the next line:

```java
tables.computeIfAbsent(tableId, k -> new HashSet<>()).addAll(ke.getValue());
```

File: core/src/main/java/org/apache/accumulo/core/clientImpl/TabletLocatorImpl.java [hunk 2]
```diff
@@ -730,17 +730,10 @@ private void processInvalidated(ClientContext context, LockCheckerSession lcSess
...
-    }
+    Map<KeyExtent,List<Range>> tablets =
+        binnedRanges.computeIfAbsent(location, k -> new HashMap<>());
 
-    List<Range> tabletsRanges = tablets.get(ke);
-    if (tabletsRanges == null) {
-      tabletsRanges = new ArrayList<>();
-      tablets.put(ke, tabletsRanges);
-    }
+    List<Range> tabletsRanges = tablets.computeIfAbsent(ke, k -> new ArrayList<>());
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/TabletLocatorImpl.java]: These look like they can simplify to:

```java
binnedRanges.computeIfAbsent(location, k -> new HashMap<>())
  .computeIfAbsent(ke, k -> new ArrayList<>())
  .add(range);
```

File: test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java [hunk 3]
```diff
@@ -213,7 +213,8 @@ private void addOpenWals(ServerContext c, Set<String> allWalsSeen) throws Except
           allWalsSeen.add(entry.getKey());
           foundWal = true;
         } else {
-          log.debug("The WalState for {} is {}", entry.getKey(), entry.getValue()); // CLOSED or UNREFERENCED
+          log.debug("The WalState for {} is {}", entry.getKey(), entry.getValue()); // CLOSED or
+                                                                                    // UNREFERENCED
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java]: This seems unrelated. If having trouble with the formatter, could just remove that comment at the end of the line, or put it above the log statement.

Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/functional/ManyWriteAheadLogsIT.java]: Disregard this one. I fixed the formatting in the master branch.

**Comment by jmark99** [comment]
@ctubbsii I added additional refactoring/chaining as you suggested. Can you take a look and verify I didn't break anything along the way. I successfully ran the unit tests and Sunny IT's.
